QModManager/Utility/MainMenuMessages.cs

```diff
@@ -0,0 +1,167 @@
+﻿namespace QModManager.Utility
+{
+    using System.Reflection;
+    using System.Reflection.Emit;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    using Harmony;
+    using UnityEngine;
+    using UnityEngine.SceneManagement;
+
+    /// <summary>
+    /// Allows to add critical messages to the main menu.
+    /// Messages will stay in the main menu and on loading screen.
+    /// </summary>
+    internal static class MainMenuMessages
+    {
+        internal const int defaultSize = 25;
+        internal const string defaultColor = "red";
+
+        private static readonly Vector2 newOffset = new Vector2(20f, 20f);
+        private static Vector2 prevOffset;
+
+        private static List<string> messageQueue;
+        private static List<ErrorMessage._Message> messages;
+
+        /// <summary>Adds an error message to the main menu.</summary>
+        /// <param name="msg">The message to add.</param>
+        /// <param name="callerID">The ID of the caller (or null for "QModManager").</param>
+        /// <param name="size">The size of the text.</param>
+        /// <param name="color">The color of the text.</param>
+        /// <param name="autoformat">Whether or not to apply formatting tags to the message, or show it as it is.</param>
+        internal static void Add(string msg, string callerID = null, int size = defaultSize, string color = defaultColor, bool autoformat = true)
+        {
+            if (Patches.hInstance == null) // just in case
+            {
+                Logger.Error($"Tried to add main menu message before Harmony was initialized. (Message: \"{msg}\")");
+                return;
+            }
+
+            Init();
+            Logger.Debug($"Created message: \"{msg}\"");
+
+            if (autoformat)
+                msg = $"<size={size}><color={color}><b>[{callerID ?? "QModManager"}]:</b> {msg}</color></size>";
+
+            if (ErrorMessage.main != null)
+                AddInternal(msg);
+            else
+                messageQueue.Add(msg);
+        }
+
+        private static void Init()
+        {
+            if (messageQueue != null)
+                return;
+
+            messageQueue = new List<string>();
+            messages = new List<ErrorMessage._Message>();
+            Patches.Patch();
+
+            SceneManager.sceneLoaded += OnSceneLoaded;
+        }
+
+        private static void AddInternal(string msg)
+        {
+            ErrorMessage.AddDebug(msg);
+
+            var message = ErrorMessage.main.GetExistingMessage(msg);
+            messages.Add(message);
+            message.timeEnd += 1e6f;
+            message.entry.rectTransform.sizeDelta = new Vector2(1920f - newOffset.x * 2f, 0f);
+        }
+
+        private static void OnSceneLoaded(Scene scene, LoadSceneMode _)
+        {
+            if (scene.name != "Main")
```

Comment:
Is that the same case in BZ?

Reply:
Hmm, good question, I don't know :) Is QMM using same codebase for both? I didn't find any separate configurations.

Reply:
Just checked and both scenes are called main though idk how yours works when all of UWE code has it in lower case.

Reply:
I don't have BZ, but in SN I see "Main" in code and in log. Where did you see lower case?

Reply:
![image](https://user-images.githubusercontent.com/7858500/80848448-8e97e880-8bd0-11ea-9f3f-a3af20faea40.png)

Reply:
Looks like both are Main   and somehow main for both games so if it works on one it should on the other.